lib/pthreads/threadpool.nit

```diff
@@ -47,20 +47,49 @@ end
 private class PoolThread
 	super Thread
 
-	var queue: ConcurrentList[Task]
+	var queue: ConcurrentList[JoinTask]
 	var mutex: Mutex
 	var cond : NativePthreadCond
 
 	redef fun main do
 		loop
-			var t: nullable Task = null
+			var t: nullable JoinTask = null
 			mutex.lock
 			if queue.is_empty then cond.wait(mutex.native.as(not null))
 			if not queue.is_empty then
-				t = queue.pop
+				t = queue.shift
 			end
 			mutex.unlock
-			if t != null then t.main
+			if t != null then
+				t.main
+				t.is_done = true
+			end
```

Comment:
It is not necessary to restrict the compatibility from `Task` to `JoinTask`, and it will make it harder and less efficient for users of simple `Task` to use a `ThreadPool`. You should keep the general type `Task` and only do something different to set `is_done`. It could be either a type check (but @privat doesn't like those), or an empty private method `Task::set_is_done` implemented only in `JoinTask`.